plugins/ZoomButtons.jsx

```diff
@@ -26,17 +29,58 @@ import './style/Buttons.css';
 class ZoomButton extends React.Component {
     static propTypes = {
         changeZoomLevel: PropTypes.func,
+        click: PropTypes.object,
+        currentTask: PropTypes.string,
         currentZoom: PropTypes.number,
         direction: PropTypes.number,
+        /** Enable zoom in or out by box selection. */
+        enableZoomByBoxSelection: PropTypes.bool,
+        mapCrs: PropTypes.string,
         mapMargins: PropTypes.object,
         maxZoom: PropTypes.number,
         /** The position slot index of the map button, from the bottom (0: bottom slot). */
         position: PropTypes.number,
+        setCurrentTask: PropTypes.func,
         theme: PropTypes.object,
         /** Omit the button in themes matching one of these flags. */
         themeFlagBlacklist: PropTypes.arrayOf(PropTypes.string),
         /** Only show the button in themes matching one of these flags. */
-        themeFlagWhitelist: PropTypes.arrayOf(PropTypes.string)
+        themeFlagWhitelist: PropTypes.arrayOf(PropTypes.string),
+        zoomToExtent: PropTypes.func,
+        zoomToPoint: PropTypes.func
+    };
+    static defaultProps = {
+        enableZoomByBoxSelection: false
+    };
+    state = {
+        disabled: false,
+        task: null,
+        zoomBox: null
+    };
+    componentDidUpdate(prevProps) {
+        if (prevProps !== this.props) {
+            if (this.props.direction > 0) {
+                this.setState({disabled: this.props.currentZoom >= this.props.maxZoom, task: "ZoomIn"});
+            } else if (this.props.direction < 0) {
+                this.setState({disabled: this.props.currentZoom <= 0, task: "ZoomOut"});
+            }
+        }
+        if (this.props.currentTask !== null && this.props.currentTask === this.state.task && this.state.disabled) {
+            this.props.setCurrentTask(null);
+        }
+        if (this.props.currentTask === this.state.task && prevProps !== this.props) {
```

Comment:
If you write `this.props.click && this.props.click !== prevProps.click` here and directly use  `this.props.click.coordinate` below you can simplify the code

Reply:
Resolved